### PR TITLE
Retarget PublicApiAnalyzer to Microsoft.CodeAnalysis 1.2.1

### PR DESCRIPTION
**Pull request by sharwell**
🚧 Being submitted as three commits, but I'm waiting for the CI to pass on each before moving to the next commit.

* [x] Split Analyzer.Utilities into Analyzer.Utilities (compiler) and Workspaces.Utilities (workspaces)
* [x] Remove workspaces dependency from analyzer-only projects
* [x] Retarget PublicApiAnalyzer to Microsoft.CodeAnalysis 1.2.1

### CODE REVIEW

File: src/Utilities/Workspaces/CodeMetrics/CodeAnalysisMetricDataExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.CodeMetrics
+{
+    internal static class CodeAnalysisMetricDataExtensions
```
Comment (drewnoakes) [inline src/Utilities/Workspaces/CodeMetrics/CodeAnalysisMetricDataExtensions.cs]: The name suggests it contains extension methods.

Comment (sharwell) [reply src/Utilities/Workspaces/CodeMetrics/CodeAnalysisMetricDataExtensions.cs]: I created the class before realizing the method I was moving was static 😊

Comment (sharwell) [reply src/Utilities/Workspaces/CodeMetrics/CodeAnalysisMetricDataExtensions.cs]: I have a solution though...

Comment (sharwell) [reply src/Utilities/Workspaces/CodeMetrics/CodeAnalysisMetricDataExtensions.cs]: Fixed

File: src/Utilities/Workspaces/xlf/WorkspacesUtilitiesResources.cs.xlf [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<xliff xmlns="urn:oasis:names:tc:xliff:document:1.2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.2" xsi:schemaLocation="urn:oasis:names:tc:xliff:document:1.2 xliff-core-1.2-transitional.xsd">
+  <file datatype="xml" source-language="en" target-language="cs" original="../AnalyzerUtilitiesResources.resx">
+    <body>
+      <trans-unit id="VersionCheckDescription">
```
Comment (mavasani) [inline src/Utilities/Workspaces/xlf/WorkspacesUtilitiesResources.cs.xlf]: These resources should no longer be used, version check analyzer has been moved to a separate project. I seemed to have forgotten to delete them from utilities.

Comment (sharwell) [reply src/Utilities/Workspaces/xlf/WorkspacesUtilitiesResources.cs.xlf]: I couldn't find the corresponding RESX file. Should I delete these XLF files altogether, or just prune them somehow?

Comment (mavasani) [reply src/Utilities/Workspaces/xlf/WorkspacesUtilitiesResources.cs.xlf]: Ah, so seems like I deleted the resx but not the xlf files. I think probably best to just delete the XLF files altogether.

Comment (sharwell) [reply src/Utilities/Workspaces/xlf/WorkspacesUtilitiesResources.cs.xlf]: I deleted the files as a fourth commit, whenever it gets there 😄

Comment (sharwell) [reply src/Utilities/Workspaces/xlf/WorkspacesUtilitiesResources.cs.xlf]: Actually I amended the first commit since the build for it hadn't even started.

**Pull request merged**